### ISSUE TITLE
Add support for STORED client side defaulted columns

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13443]
In jOOQ 3.17, client side computed columns (both `STORED` and `VIRTUAL`) have been introduced with https://github.com/jOOQ/jOOQ/issues/9879.

Throughout the discussion there, the option of doing this also with a `DEFAULT` / `STORED` semantics has been discussed, but was postponed, as the feature is less interesting than computed columns (`GENERATED ALWAYS AS`)

### Tasks

- [ ] Code generation support
- [ ] Runtime support

Task list will be updated.

**Comment by alf** [comment]
We have a view that filters on a column value. It would be nice to set a default on that column to the filtered value.

**Comment by lukaseder** [comment]
So, e.g. 

```sql
CREATE VIEW v (a, b)
AS
SELECT a, b
FROM t
WHERE b = 1;
```

And now you'd like to make sure that when inserting into this view, `b` is 1 by default, is that it? You probably also have a `WITH CHECK OPTION` clause to ensure this is the case, but it would be convenient if users could omit the `b` column from their `INSERT` statements.

**Comment by alf** [comment]
Exactly. We can handle this with an `instead of`-trigger in the database but this has two major downsides:

1. The instead of trigger is a pain to maintain due to a lot of boilerplate
2. The instead of trigger causes the RETURNING-functionality to stop working when doing inserts into the view

**Comment by lukaseder** [comment]
Btw: Upvoting the issue itself helps with prioritisation. I frequently look at this report to see what should be done next:
https://github.com/jOOQ/jOOQ/issues?page=1&q=is%3Aissue+sort%3Areactions-%2B1-desc+is%3Aopen

**Comment by alf** [comment]
Sorry for not getting back to you here. The real world example is about as simple as yours. However we're not able to solve this with a STORED computed column without adding some mechanism for marking this column as dirty/mentioned in the insert-statement. If we add this mechanism we might as well just set the default value there.

Maybe I'm missing something?

**Comment by lukaseder** [comment]
I'm not sure if *I'm* missing something. Why do you need to make it dirty?

**Comment by alf** [comment]
Because we want the column to be populated and written to the database. We populate an updatable record and store. Unless the calculated column gets touched it’s left out from the resulting DML.

**Comment by lukaseder** [comment]
> Unless the calculated column gets touched it’s left out from the resulting DML.

Hmm, no, the point of it being calculated (and stored), is that it must be calculated, always, irrespective of what you do with your `UpdatableRecord`. Do you have a specific set of API calls that produces a bug?
